src/zenml/artifact_stores/base_artifact_store.py

```diff
@@ -106,6 +110,55 @@ def inner_function(*args: Any, **kwargs: Any) -> Any:
     return inner_function
 
 
+def _verify(_func: Callable[..., Any]) -> Callable[..., Any]:
+    """Verifies that the file belongs to the artifact store.
+
+    Args:
+        _func: The function for which to verify the inputs.
+
+    Returns:
+        Function that calls the input function.
+    """
+
+    def inner_function(
+        self: "BaseArtifactStore", *args: Any, **kwargs: Any
+    ) -> Any:
```

Comment:
You don't need this complication at all if you just reuse the `_sanitize_paths` and `_sanitize_potential_path` and modify them to accept a `root_path` argument instead.

Reply:
reworked that part to unify in one deco only